### ISSUE TITLE
请问，有没有打算提供更好的可访问性支持 （accessibility support）？

### ISSUE THREAD

**Issue by yccheok** [huanghaibin-dev/CalendarView#510]
嗨，今天我收到一封，来自有视觉障碍的用户。

> 
> Greetings,
> 
>  
> 
> I would like to start off by saying that I came across “???” today and from its description, it sounded exactly like what I am after.
> 
>  
> 
> I installed the app, and as a blind user who relies on a screen Reader, I must say that just about everything falls in line with Google’s guidelines for developing with accessibility in mind, thus, I get feedback from just about every element or controls on the app except for the fact that the dates are not exposed to Talkback, the Screen Reader that comes built-in on android phones.
> 
>  
> 
> Being an XDA person that you are, I am more than confident that you would fix this issue and open up the app to be fully enjoyed by blind folks like me.
> 
>  
> 
> I look forward to hearing from you  regarding this issue as I hate leaving negative ratings on the Play Store without raising an issue with the Dev offline.
> 
>  
> 
> Thank you again for your great work and I can’t wait to introduce the app to an android group for blind people around the world where we discuss apps that are accessible and those to avoid.
> 
> Of course, I want this one to go to the accessible column and not the one to avoid.
> 
>  
> 
> Thank you.
> 
>  
> 
> Sincerely,
> ???

我的回复如下

> Hi ???,
> 
> If I guess correctly, you are referring to "date" at Calendar page? You wish 1st, 2nd, 3rd, ... 31st date to be spelled out, as you select them?
> 
> If that is so, it will take some time to fix them. Let me explain why.
> 
> 1) Talkback works best, when developers are using standard built-in elements such as Button, TextView, ... 
> 
> 2) However, for this Calendar page it is rather special. It is implemented by not using standard built-in element. It is implemented as "Drawing on canvas". Talkback doesn't work out-of-box with custom drawing. The reason for using "Drawing on canvas" is simple - speed. You will notice that our Calendar page is faster the other similar Calendar page in the market.
> 
> 3) I have no technical knowledge on Talkback. Hence, I need to take some time to study, and implement it. Moreover, this Calendar is provided by a 3rd party open source developer. I need to spend considerable amount of time, to communicate and work together with him.
> 
> Thank you.

这看来似乎是很有意义的功能。

不知，贵软件是否有意，在不影响性能之下，开发这功能？我也很乐意提供 patch，虽然我对talkback完全不熟悉。

这是些技术文档 － https://developer.android.com/guide/topics/ui/accessibility/custom-views
